### PR TITLE
Make roads be created between buildings

### PR DESCRIPTION
**Pull request by BartekBanas**
Implemented RoadManger that creates and updates roads upon placing a new building

### CODE REVIEW

File: TileTowns/Assets/Scripts/Managers/RoadManager.cs [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    [SerializeField] public Tile upLeftRoad;
+    [SerializeField] public Tile leftRightRoad;
+    [SerializeField] public Tile upRoad;
+    [SerializeField] public Tile rightRoad;
+    [SerializeField] public Tile downRoad;
+    [SerializeField] public Tile leftRoad;
+    
+    private Tilemap _tilemap;
+    private Dictionary<Vector2Int, bool> _roadMap = new();
+    private const int RoadZIndex = 1;
```
Comment (MountainCat1) [inline TileTowns/Assets/Scripts/Managers/RoadManager.cs:33]: consts are supposed to be at the top of the class

File: TileTowns/Assets/Scripts/Managers/RoadManager.cs [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+    
+    private void Start()
+    {
+        _gameManager.LevelLoaded += InitializeRoadMap;
+        _gameManager.LevelLoaded += OnGameLoaded;
+        
+        RoadPlaced += CreateRoad;
+        _buildingController.PlacedBuilding += (_, tileData) => RoadPlaced?.Invoke(tileData);
```
Comment (MountainCat1) [inline TileTowns/Assets/Scripts/Managers/RoadManager.cs:41]: you sure its not better to just make a method OnBuildingPlaced()?
this doesnt seem too readable

File: TileTowns/Assets/Scripts/Managers/RoadManager.cs [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+        {
+            SetRoad(position, leftRoad);
+        }
+        else if (hasRoadRight)
+        {
+            SetRoad(position, rightRoad);
+        }
+        else
+        {
+            // No roads around
```
Comment (MountainCat1) [inline TileTowns/Assets/Scripts/Managers/RoadManager.cs:165]: you can add InvalidOperationException or something (optional)

File: TileTowns/Assets/Scenes/GameScene.unity [hunk 4]
```diff
@@ -742,6 +742,65 @@ RectTransform:
   m_AnchoredPosition: {x: 0, y: 0}
   m_SizeDelta: {x: 0, y: 0}
   m_Pivot: {x: 0, y: 0}
+--- !u!1 &226715574
```
Comment (MountainCat1) [inline TileTowns/Assets/Scenes/GameScene.unity:745]: i wont bother checking how its in a scene but i hope you just made a new object called RoadManger or something along side other managers added a script to it ;3

**Review by MountainCat1 — CHANGES_REQUESTED** [review]

File: TileTowns/Assets/Scripts/Managers/RoadManager.cs (hunk 3 above)
Comment (BartekBanas) [reply TileTowns/Assets/Scripts/Managers/RoadManager.cs:165]: It's just a scenario, where it'a a lone building, don't think it should throw any exception

File: TileTowns/Assets/Scripts/Managers/RoadManager.cs (hunk 2 above)
Comment (MountainCat1) [reply TileTowns/Assets/Scripts/Managers/RoadManager.cs:41]: Events are for communicating between objects, not inside them ;)

File: TileTowns/Assets/Scripts/Managers/RoadManager.cs (hunk 3 above)
Comment (MountainCat1) [reply TileTowns/Assets/Scripts/Managers/RoadManager.cs:165]: ow wait fuck xD
didnt read into it, you can write the `intentionally empty`, thats quite a good practice but however you prefer ;)

**Pull request merged**